store/dbInstance.go

```diff
@@ -55,94 +56,99 @@ type DBInstance struct {
 	Processor     string `json:"processor"`
 }
 
-// Convert convert the client api message to the storage form
-func Convert(priceList []*client.Offer, instanceList []*client.Instance) ([]*DBInstance, error) {
-	var dbInstanceList []*DBInstance
-	dbInstanceMap := make(map[string]*DBInstance)
-	offerMap := make(map[string][]*Term)
-
-	for _, offer := range priceList {
-		var payload *TermPayload
+// convertAWS convert the offer provided by AWS to DBInstance
+func convertAWS(offerList []*client.Offer) ([]*DBInstance, error) {
+	termMap := make(map[int][]*Term)
+	for _, offer := range offerList {
+		// filter the offer does not have a instancePayload (only got price but no goods).
+		if offer.InstancePayload == nil {
+			continue
+		}
+		var termPayload *TermPayload
 		// Only reserved type has payload field
 		if offer.ChargeType == client.ChargeTypeReserved {
-			payload = &TermPayload{
+			termPayload = &TermPayload{
 				LeaseContractLength: offer.ChargePayload.LeaseContractLength,
 				PurchaseOption:      offer.ChargePayload.PurchaseOption,
 			}
 		}
 
 		term := &Term{
-			Type:        offer.ChargeType,
-			Payload:     payload,
-			Unit:        offer.Unit,
-			USD:         offer.USD,
-			Description: offer.Description,
+			DatabaseEngine: offer.InstancePayload.DatabaseEngine,
+			Type:           offer.ChargeType,
+			Payload:        termPayload,
+			HourlyUSD:      offer.HourlyUSD,
+			CommitmentUSD:  offer.CommitmentUSD,
 		}
-		if termList, ok := offerMap[offer.InstanceID]; !ok {
-			offerMap[offer.InstanceID] = []*Term{term}
+		if _, ok := termMap[offer.ID]; !ok {
+			termMap[offer.ID] = []*Term{term}
 		} else {
-			offerMap[offer.InstanceID] = append(termList, term)
+			termMap[offer.ID] = append(termMap[offer.ID], term)
 		}
 	}
 
 	now := time.Now().UTC()
 	incrID := 0
-	for _, instance := range instanceList {
+	// dbInstanceMap is used to aggregate the instance by their type (e.g. db.m3.large).
+	dbInstanceMap := make(map[string]*DBInstance)
+	var dbInstanceList []*DBInstance
+	// extract dbInstance from the payload field stored in the offer.
+	for _, offer := range offerList {
+		// filter the offer does not have a instancePayload (only got price but no goods).
```

Comment:
nit: "filter the offer does not have a instancePayload" doesn't say a thing because "if offer.InstancePayload == nil" below already says it. We should comment on when or how offer.InstancePayload could be nil.